File: design-patterns/java/chainofresponsibility/AuthHandler.java
public class AuthHandler extends BaseHandler {
    @Override
    public void handle(Request request) {
        if (request.user == null) {
            System.out.println("AuthHandler: ❌ User not authenticated.");
            return; // Stop the chain
        }
        System.out.println("AuthHandler: ✅ Authenticated.");
        forward(request);
    }
}

File: design-patterns/csharp/chainofresponsibility/AuthHandler.cs
using System;

namespace ChainOfResponsibility
{
    public class AuthHandler : BaseHandler
    {
        public override void Handle(Request request)
        {
            if (string.IsNullOrEmpty(request.User))
            {
                Console.WriteLine("AuthHandler: User is not authenticated");
                return;
            }

            Console.WriteLine($"AuthHandler: User {request.User} is authenticated");
            HandleNext(request);
        }
    }
} 